drivers/media/i2c/mira016.inl

```diff
@@ -0,0 +1,6501 @@
+	{30, 0},
+	{31, 2},
+	{43, 0},
+	{57348, 1},
+	{30, 0},
+	{31, 2},
+	{43, 0},
+	{57344, 0},
+	{57344, 0},
+	{31, 0},
+	{32, 0},
+	{35, 0},
+	{36, 1},
+	{37, 144},
+	{38, 0},
+	{39, 14},
+	{40, 0},
+	{41, 1},
+	{42, 144},
+	{43, 0},
+	{44, 14},
+	{45, 0},
+	{46, 0},
+	{47, 0},
+	{48, 0},
+	{49, 0},
+	{50, 0},
+	{51, 0},
+	{52, 0},
+	{53, 0},
+	{54, 0},
+	{55, 0},
+	{56, 0},
+	{57, 0},
+	{58, 0},
+	{59, 0},
+	{60, 0},
+	{61, 0},
+	{62, 0},
+	{63, 0},
+	{64, 0},
+	{65, 0},
+	{66, 0},
+	{67, 0},
+	{68, 0},
+	{69, 0},
+	{70, 0},
+	{71, 0},
+	{72, 0},
+	{73, 0},
+	{74, 0},
+	{75, 0},
+	{76, 0},
+	{77, 0},
+	{78, 0},
+	{79, 0},
+	{80, 0},
+	{81, 0},
+	{82, 0},
+	{83, 0},
+	{84, 0},
+	{85, 0},
+	{57344, 1},
+	{57348, 0},
+	{57344, 1},
+	{57388, 0},
+	{57389, 14},
+	{57390, 1},
+	{57391, 157},
+	{57392, 0},
+	{57381, 0},
+	{57386, 0},
+	{8233, 40},
+	{52, 0},
+	{53, 200},
+	{57348, 0},
+	{30, 0},
+	{31, 2},
+	{43, 0},
+	// { 57348, 0},
+	// { 14, 0},
+	// { 15, 0},
+	// { 16, 3},
+	// { 17, 232},
+	// { 18, 0},
+	// { 19, 0},
+	// { 20, 0},
+	// { 21, 0},
+	{57348, 0},
+	{50, 4},
+	{51, 18},
+	{57348, 0},
+	{7, 1},
+	// { 8, 0},
+	// { 9, 0},
+	// { 10, 10},
+	// { 11, 217},
+	{57348, 0},
+	{49, 0},
+	{57348, 0},
+	{38, 0},
+	{57348, 0},
+	// { 28, 0},
+	// { 25, 0},
+	// { 26, 7},
+	// { 27, 83},
+	// { 22, 8},
+	// { 23, 0},
+	// { 24, 0},
+	{57348, 0},
+	// { 29, 0},
+	{57344, 0},
+	{57344, 0},
+	{58156, 1},
+	{57344, 0},
+	{58157, 1},
+	{57344, 0},
+	{57678, 0},
+	{57344, 0},
+	{58130, 127},
+	{57344, 0},
+	{58153, 3},
+	{57344, 0},
+	{58155, 0},
+	{57344, 0},
+	{58161, 15},
+	{57344, 0},
+	{58162, 0},
+	{57344, 0},
+	{58154, 0},
+	{57344, 0},
+	{57630, 1},
+	{0, 0},
+	{0, 0},
+	{434, 1},
+	{435, 28},
+
+
+
+	{0xE000, 0},
+	{0xE004, 0},
+	{0x33D, 1},
+};
+
+static const struct mira016_reg full_400_400_100fps_8b_1lane_reg_post_soft_reset[] = {
+	{0xE000, 0},
+	{0xE004, 0},
+	// Below are manually added after reg seq txt
+	{0x0335, 1},  // iref sel
+	{0x0324, 43}, // iref val
+
+	{0x1d9, 1},	 // #vddana sel
+	{0x0EB, 15}, // #vddana val trim
+
+	{0x1dd, 1},	 // # vsspc sel
+	{0x1ed, 14}, // # vsspc val
+
+	{0x1df, 1}, // #cp sel
+	{0x0ee, 4}, // #cp trim,
+};
+
+static const char *const mira016_test_pattern_menu[] = {
+	"Disabled",
+	"Fixed Data",
+	"2D Gradient",
+};
+
+static const int mira016_test_pattern_val[] = {
+	MIRA016_TEST_PATTERN_DISABLE,
+	MIRA016_TEST_PATTERN_FIXED_DATA,
+	MIRA016_TEST_PATTERN_2D_GRADIENT,
+};
+
+/* regulator supplies */
+static const char *const mira016_supply_name[] = {
+	// TODO(jalv): Check supply names
+	/* Supplies can be enabled in any order */
+	"VANA", /* Analog (2.8V) supply */
+	"VDIG", /* Digital Core (1.8V) supply */
+	"VDDL", /* IF (1.2V) supply */
+};
+
+#define MIRA016_NUM_SUPPLIES ARRAY_SIZE(mira016_supply_name)
+
+/*
+ * The supported formats. All flip/mirror combinations have the same byte order because the sensor
+ * is monochrome
+ */
+static const u32 codes[] = {
+	MEDIA_BUS_FMT_SGRBG10_1X10,
```

Comment:
Why are you using Bayer colour filter formats if the sensor is mono? MEDIA_BUS_FMT_Y10_1X10.